spec/lib/jobs/ucs-service-profile-discovery-spec.js

```diff
@@ -0,0 +1,146 @@
+            uri:'https://1.1.1.1:7080/sys',
+            username:'user',
+            password:'pass',
+            ucs:"1.1.1.1"
+        }, {}, graphId);
+        
+        sandbox.stub(ucsJob.ucs);
+        ucsTool = ucsJob.ucs;
+
+        rootData = {
+            body: {
+                ServiceProfile: {
+                    members:[
+                        {
+                            "assoc_state": "associated",
+                            "associatedServer": "sys/chassis-3/blade-3",
+                            "name": "ls-TTTTTTTTTT",
+                            "path": "org-root/ls-TTTTTTTTTT"
+                        },
+                        {
+                            "assoc_state": "associated",
+                            "associatedServer": "sys/rack-unit-3",
+                            "name": "ls-Profile3",
+                            "path": "org-root/ls-Profile3"
+                        },
+                        {
+                            "assoc_state": "associated",
+                            "associatedServer": "sys/chassis-4/blade-1",
+                            "name": "ls-SP4",
+                            "path": "org-root/ls-SP4"
+                        },
+                        {
+                            "assoc_state": "associated",
+                            "associatedServer": "sys/rack-unit-1",
+                            "name": "ls-SP3",
+                            "path": "org-root/ls-SP3"
+                        },
+                        {
+                            "assoc_state": "associated",
+                            "associatedServer": "sys/rack-unit-2",
+                            "name": "ls-SP2",
+                            "path": "org-root/ls-SP2"
+                        }
+                    ],
+                    org: "root"
+                }
+            }
+        };
+        waterline.nodes.updateOne.reset();
+    });
+    
+    describe('usc service profile discovery', function() {
+        it('should successfully run the job', function() {
+            ucsTool.clientRequest.onCall(0).resolves(rootData);
+            ucsTool.clientRequest.onCall(1).resolves(rootData);
+            return ucsJob._run()
+                .then(function() {
+                     expect(waterline.nodes.updateOne).to.be.called.once;
+                });
+        });
+
+        it('should not update a node', function() {
+            ucsTool.clientRequest.onCall(0).resolves([]);
+            return ucsJob._run()
+                .then(function() {
+                    expect(waterline.nodes.updateOne).to.not.be.called.once;
+                });
+        });
+
+
+        it('should successfully log in', function() {
+            ucsTool.clientRequest.onCall(0).resolves({"body":"cookie"});
+            return ucsJob.logIn()
+                .then(function(data) {
+                    expect(data).to.deep.equal("cookie");
+                });
+        });
+    });
+});
```

Comment:
I don't think the test coverage is good enough, I see a lot logic is actually not tested, for example:
(1) If ucs request error, then the job should fail.
(2) If input any invalid option, the job should fail.
(3) If ucs request returns multiple servers, then multiple nodes should be updated or created.
(4) The obm service should be created (or updated) when the node is created (or updated)